### ISSUE TITLE
Allow disabling MultiSourceReader trackData from the standard parser flow

### ISSUE THREAD

**Issue by AntaresS** [graphql-java/graphql-java#3061]
Currently, from the standard [ParseAndValidate](https://github.com/graphql-java/graphql-java/blob/fec817518772e0bcc1cdce1ee10193fe659973fd/src/main/java/graphql/ParseAndValidate.java#L74) flow it doesn't provide any options to configure the underlying [MultiSourceReader](https://github.com/graphql-java/graphql-java/blob/fec817518772e0bcc1cdce1ee10193fe659973fd/src/main/java/graphql/parser/Parser.java#L261-L264) because the document set in the [ParserEnviroment](https://github.com/graphql-java/graphql-java/blob/fec817518772e0bcc1cdce1ee10193fe659973fd/src/main/java/graphql/ParseAndValidate.java#L71) is always from a string so that it builds the MultiSourceReader with defaults.

This will make the MultiSourceReader not configurable especially the [`trackData`](https://github.com/graphql-java/graphql-java/blob/fec817518772e0bcc1cdce1ee10193fe659973fd/src/main/java/graphql/parser/MultiSourceReader.java#L224) option will always be true which creates memory allocation even if the tracked source locations are not needed. Can we introduce a new `trackData` option in the `ParserOptions` to allow users who want to stick with the standard parser flow but also want to configure it?

```
multiSourceReader = MultiSourceReader.newMultiSourceReader()
                    .reader(reader, null)
                    .trackData(environment.getParserOptions().isTrackData())
                    .build();
```

**Comment by bbakerman** [comment]
I agree with this and have approved the PR

**Comment by dondonz** [comment]
Closed by #3062, thanks!
